devicetypes/smartthings/zwave-water-temp-light-sensor.src/zwave-water-temp-light-sensor.groovy

```diff
@@ -0,0 +1,226 @@
+	// Dome Leak Sensor sends WakeUpNotification every 12 hours. Please add zwaveinfo.mfr check when adding other sensors with different interval.
+	sendEvent(name: "checkInterval", value: (2 * 12 + 2) * 60 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+}
+
+private getCommandClassVersions() {
+	[0x20: 1, 0x30: 1, 0x31: 5, 0x80: 1, 0x84: 1, 0x71: 3, 0x9C: 1]
+}
+
+def parse(String description) {
+	def result = null
+	if (description.startsWith("Err")) {
+		result = createEvent(descriptionText: description)
+	} else {
+		def cmd = zwave.parse(description, commandClassVersions)
+		if (cmd) {
+			result = zwaveEvent(cmd)
+		} else {
+			result = createEvent(value: description, descriptionText: description, isStateChange: false)
+		}
+	}
+	log.debug "Parsed '$description' to $result"
+	return result
+}
+
+def sensorValueEvent(value) {
+	def eventValue = value ? "wet" : "dry"
+	createEvent(name: "water", value: eventValue, descriptionText: "$device.displayName is $eventValue")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	sensorValueEvent(cmd.value)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicSet cmd) {
+	sensorValueEvent(cmd.value)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd) {
+	sensorValueEvent(cmd.value)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensorbinaryv1.SensorBinaryReport cmd) {
+	sensorValueEvent(cmd.sensorValue)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensoralarmv1.SensorAlarmReport cmd) {
+	sensorValueEvent(cmd.sensorState)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	def result = []
+	if (cmd.notificationType == 0x05) {
+		switch (cmd.event) {
+			case 0x00:
+				if (cmd.eventParametersLength && cmd.eventParameter.size() && eventParameter[0] > 0x02) {
+					result << createEvent(descriptionText: "Water alarm cleared", isStateChange: true)
+				} else {
+					result << createEvent(name: "water", value: "dry")
+				}
+				break
+			case 0xFE:
+				result << createEvent(name: "water", value: "dry")
+				break
+			case 0x01:
+			case 0x02:
+				result << createEvent(name: "water", value: "wet")
+				break
+			case 0x03:
+			case 0x04:
+				result << createEvent(descriptionText: "Water level dropped", isStateChange: true)
+				break
+			case 0x05:
+				result << createEvent(descriptionText: "Replace water filter", isStateChange: true)
+				break
+			case 0x06:
+				def level = ["alarm", "alarm", "below low threshold", "above high threshold", "max"][cmd.eventParameter[0]]
+				result << createEvent(descriptionText: "Water flow $level", isStateChange: true)
+				break
+			case 0x07:
+				def level = ["alarm", "alarm", "below low threshold", "above high threshold", "max"][cmd.eventParameter[0]]
+				result << createEvent(descriptionText: "Water pressure $level", isStateChange: true)
+				break
+		}
+	} else if (cmd.notificationType == 0x04) {
```

Comment:
Does that device report overheat?
Maybe consider using Temperature Alarm capability?

Reply:
No. It doesn't.

But I figured it might be worth having this stuff for some hypothetical device in the future.